tests/e2e/step_definitions/hooks.js

```diff
@@ -0,0 +1,28 @@
+const { defineSupportCode } = require('cucumber');
+const state = require('../pop/index');
+const expect = require('chai').expect;
+
+defineSupportCode(({ Before, After, setDefaultTimeout }) => {
+    setDefaultTimeout(60 * 1000);
+
+    Before({tags: '@login'}, () => {
+        browser.url(browser.options.baseUrl + state.getPageByName('Home').pageUrl);
+        state.getPage().getElement('Menu -> Login').click();
+        browser.pause(2000);
+        expect(browser.getUrl()).to.eql(browser.options.baseUrl + state.getPageByName('Login').pageUrl);
+        state.getPage().getElement('Username').setValue('maksimliubinski');
+        state.getPage().getElement('Password').setValue('npmjspassword');
+        state.getPage().getElement('Login').click();
+        expect(browser.getUrl()).to.eql(browser.options.baseUrl + state.getPageByName('Home').pageUrl);
```

Comment:
IMHO, hooks should not include any checks, so for this case better to add waiter that url is changed or element is visible (line 24-25)

Reply:
Agree

Reply:
You can use cli for user and password, browser.options.memory.getValue(credentials.user) for getting credentials